### DISCUSSION TITLE
用自己的数据集进行tsne可视化分析

### DISCUSSION THREAD

**Discussion by huadouxiansheng** [open-mmlab/mmselfsup#710] [How-to]
如题用自己的数据集写 t-SNE 可视化的数据配置文件，需要修改哪些配置，还是目前不支持imagenet数据集外的数据集
我尝试修改
dataset_type = 'mmcls.CustomDataset'
data_root = 'data/custom_dataset/'
file_client_args = dict(backend='disk')
name = 'custom_val'

extract_pipeline = [
    dict(type='LoadImageFromFile', file_client_args=file_client_args),
    dict(type='mmcls.ResizeEdge', scale=256, edge='short'),
    dict(type='CenterCrop', crop_size=224),
    dict(type='PackSelfSupInputs'),
]

extract_dataloader = dict(
    batch_size=8,
    num_workers=4,
    dataset=dict(
        type=dataset_type,
        data_root='data/custom_dataset/val',
       # ann_file='meta/val.txt',
        data_prefix='val',
        pipeline=extract_pipeline),
    sampler=dict(type='DefaultSampler', shuffle=False),
)

# pooling cfg
pool_cfg = dict(type='MultiPooling', in_indices=(1, 2, 3, 4))

但是显然有错

usage: visualize_tsne.py [-h] [--checkpoint CHECKPOINT] [--work-dir WORK_DIR]
                         [--launcher {none,pytorch,slurm,mpi}]
                         [--dataset-config DATASET_CONFIG]
                         [--max-num-class MAX_NUM_CLASS] [--seed SEED]
                         [--deterministic]
                         [--cfg-options CFG_OPTIONS [CFG_OPTIONS ...]]
                         [--n-components N_COMPONENTS]
                         [--perplexity PERPLEXITY]
                         [--early-exaggeration EARLY_EXAGGERATION]
                         [--learning-rate LEARNING_RATE] [--n-iter N_ITER]
                         [--n-iter-without-progress N_ITER_WITHOUT_PROGRESS]
                         [--init INIT]
                         config
visualize_tsne.py: error: unrecognized arguments: --dataset_config configs/benchmarks/classification/tsne_custom.py

**Comment by fangyixiao18** [comment]
https://mmselfsup.readthedocs.io/en/1.x/user_guides/visualization.html#visualize-t-sne
this is the newest version of tsne
